src/controllers/krate/search.rs

```diff
@@ -92,9 +93,25 @@ pub async fn search(app: AppState, req: Parts) -> AppResult<Json<Value>> {
                 query = query.order(Crate::with_name(q_string).desc());
 
                 if sort == "relevance" {
-                    let q = sql::<TsQuery>("plainto_tsquery('english', ")
+                    // If the query string is not a stop word, search using `plainto_tsquery(...)`.
+                    // Else if the it is a valid keyword, search by casting it to `tsquery` with weight B(keyword).
+                    // Otherwise, search using `null::tsquery`.
+                    let qs = sql::<TsQuery>("plainto_tsquery('english', ")
                         .bind::<Text, _>(q_string)
                         .sql(")");
```

Comment:
Note: This could also be refactored to `plainto_tsquery_with_search_config`, which I've implemented upstream in https://github.com/diesel-rs/diesel_full_text_search/pull/41. I incorrectly used `to_tsquery_with_search_config` in the #7941  , which is not equivalent to `plainto_tsquery` and caused an issue #8052 . And since carol has concerns about it so I haven't modified it in this PR.